composer/callbacks/image_monitor.py

```diff
@@ -0,0 +1,194 @@
+            Default: ``8``.
+        shift_targets (bool): For use when ``mode="segmentation"``. Optionally shifts the target classes up by one to
+            allow for ignored/background classes with negative values. If ``shift_targets=True``, all class ids will be
+            incremented by 1, and any pixels with negative target indices will be assigned a target index of zero.
+            Default: ``False``.
+        input_key (str | int | Tuple[Callable, Callable] | Any, optional): A key that indexes to the input
+            from the batch. Can also be a pair of get and set functions, where the getter
+            is assumed to be first in the pair.  The default is 0, which corresponds to any sequence, where the first
+            element is the input. Default: ``0``.
+        target_key (str | int | Tuple[Callable, Callable] | Any, optional): A key that indexes to the target
+            from the batch. Can also be a pair of get and set functions, where the getter
+            is assumed to be first in the pair. The default is 1, which corresponds to any sequence, where the second
+            element is the target. Default: ``1``.
+    """
+
+    def __init__(self,
+                 interval: str = '100ba',
+                 mode: str = 'input',
+                 num_images: int = 8,
+                 shift_targets: bool = False,
+                 input_key: Union[str, int, Tuple[Callable, Callable], Any] = 0,
+                 target_key: Union[str, int, Tuple[Callable, Callable], Any] = 1):
+        self.interval = interval
+        self.mode = mode
+        self.num_images = num_images
+        self.shift_targets = shift_targets
+        self.input_key = input_key
+        self.target_key = target_key
+
+        try:
+            import wandb
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='wandb',
+                                                conda_package='wandb',
+                                                conda_channel='conda-forge') from e
+        del wandb  # unused
+
+        # Check that the output mode is valid
+        if self.mode.lower() not in ['input', 'segmentation']:
+            raise ValueError(f'Invalid mode: {mode}')
+
+        # Check that the interval timestring is parsable and convert into time object
+        try:
+            self.interval = Time.from_timestring(interval)
+        except ValueError as error:
+            raise ValueError(f'Invalid time string for parameter interval') from error
+
+        # Verify that the interval has supported units.
+        if self.interval.unit not in [TimeUnit.BATCH, TimeUnit.EPOCH]:
+            raise ValueError(f'Invalid time unit for parameter interval: '
+                             f'{self.interval.unit}')
+
+    def _log_inputs(self, state: State, logger: Logger, key: str):
+        inputs = state.batch_get_item(key=self.input_key)
+        if not isinstance(inputs, torch.Tensor):
```

Comment:
This check might because `_make_input_images` does `images = inputs[0:num_images].data.cpu().permute(0, 2, 3, 1).numpy()` which would only be possible for torch tensors. How desirable would it be to have PIL images and numpy arrays separated? It would add some checking to `_make_input_images`, right?

Reply:
This check might because `_make_input_images` does `images = inputs[0:num_images].data.cpu().permute(0, 2, 3, 1).numpy()` which would only be possible for torch tensors. How desirable would it be to have PIL images and numpy arrays separated? It would add some checking to `_make_input_images`, right?

Reply:
Oh this is interesting. Do we ever expect it to be the case that an input or output to the network will not be a torch tensor on BEFORE_FORWARD or BEFORE_LOSS? A tensor conversion has to happen by the time all state changes have finished on BEFORE_FORWARD or the network will crash, right? Or, is there some case where one might handle the conversion to tensor in the forward function itself?

Reply:
Thats a good point, thanks for bringing it up! I'll add a check, and then in another PR I can add the support for PIL and numpy images.